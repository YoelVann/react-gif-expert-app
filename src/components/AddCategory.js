import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState("");
    const handleInputChange = (e) =>{
        
        setinputValue(e.target.value)
    }

    const handleSubmmit = (e) =>{
        e.preventDefault();
        if( inputValue.trim().length>2){
            setCategories(cats => [inputValue,...cats]);
            setinputValue('');
        }
    }
  return (
      <form onSubmit={ handleSubmmit }>
        <h1>{inputValue}</h1>
        <input 
            type="text"
            placeholder="Escribe una categorÃ­a Ej: 'Black Clover' y presiona Enter"
            value={inputValue}
            onChange={handleInputChange}
        />
      </form>
  )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
