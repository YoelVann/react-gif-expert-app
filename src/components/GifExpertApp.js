import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    // const [categories, setCategories] = useState(['Naruto']);
    const [categories, setCategories] = useState(defaultCategories);
    
    // const handleAdd = () =>{
    //     setCategories([...categories, 'Hunter X Hunter']);
    // }

    return ( 
        <>
        <h2>Gif Expert App</h2>
        <AddCategory setCategories={setCategories}/>
        <hr></hr>
        <ol>
            {
                categories.map( category => 
                    <GifGrid 
                        key={category}
                        category={category}
                    />
                )    
            }
        </ol>
    
        </>
    );
};

export default GifExpertApp;
