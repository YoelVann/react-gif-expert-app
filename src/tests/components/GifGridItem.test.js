import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => { 
    const title = 'title';
    const url = 'www.joel.com' 
    const wrapper = shallow( <GifGridItem  title={title} url={url}/>);

    test('Debe de mostrar el componente', () => { 
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de tener un pÃ¡rrafo con el title', () => { 
        const p = wrapper.find('p');
        expect( p.text().trim()).toBe(title);
     });

    test('Debe de tener la imagen igual al url y alt de los props', () => { 
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    }); 

    test('Debe tener animate__fadeIn', () => { 
        const div = wrapper.find('div');

        // console.log(div.hasClass('animate__fadeIn'))
        expect(div.hasClass('animate__fadeIn')).toBe( true );
    });
 });
