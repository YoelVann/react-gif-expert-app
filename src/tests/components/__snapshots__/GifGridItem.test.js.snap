// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de mostrar el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title"
    src="www.joel.com"
  />
  <p>
    title
  </p>
</div>
`;
